src/itwinai/torch/inference.py

```diff
@@ -42,10 +51,11 @@ def __call__(self) -> nn.Module:
         """
         if os.path.exists(self.model_uri):
             # Model is on local filesystem.
-            model = torch.load(self.model_uri)
-            return model.eval()
+            device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
```

Comment:
If this is to be done distributed, would you not need a more sophisticated way of getting the specific device? I might be wrong here